### PR TITLE
open source companion to the pro session notifications PR

### PR DESCRIPTION
**Pull request by kfeinauer**

### CODE REVIEW

File: src/cpp/core/DateTime.cpp [hunk 1]
```diff
@@ -96,6 +100,63 @@ std::string millisecondsSinceEpochAsString(double ms)
...
+}
+
+bool parseUtcTimeFromFormatString(const std::string& timeStr,
+                                  const std::string& formatStr,
+                                  boost::posix_time::ptime *pOutTime)
+{
+   using namespace boost::local_time;
+
+   std::stringstream ss(timeStr);
+   local_time_input_facet* ifc = new local_time_input_facet(formatStr);
```
Comment (jmcphers) [inline src/cpp/core/DateTime.cpp]: Does this leak?

File: src/cpp/core/DateTime.cpp [hunk 2]
```diff
@@ -96,6 +100,63 @@ std::string millisecondsSinceEpochAsString(double ms)
...
+   return now - utc_now;
+}
+
+std::string getUtcOffsetString()
+{
+   using namespace boost::posix_time;
+
+   std::stringstream out;
+
+   time_facet* tf = new time_facet();
```
Comment (jmcphers) [inline src/cpp/core/DateTime.cpp]: Also a possible leak

Comment (jjallaire) [inline src/cpp/core/DateTime.cpp]: I'm pretty sure this isn't a leak (I think facets might get automatically deleted by the output stream). Worth digging into to make sure though!

File: src/cpp/core/DateTime.cpp (hunk 1 above)
Comment (kfeinauer) [inline src/cpp/core/DateTime.cpp]: It does not leak. The facets are managed by the stream per @jjallaire. This is just a quirkiness of the library here. If you free it you will get a crash due to double free.

Comment (jmcphers) [inline src/cpp/core/DateTime.cpp]: Thanks, I wasn't familiar with this idiom!

**Pull request merged**